### PR TITLE
mu: fix HEAD build failure due to miss a dependency

### PR DESCRIPTION
**Pull request by junkw**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
`mu` at HEAD depends on `autoconf-archive`, so I added it as a build dependency.

See https://github.com/djcb/mu/issues/975

### CODE REVIEW

File: Formula/mu.rb [hunk 1]
```diff
@@ -15,6 +15,7 @@ class Mu < Formula
   end
 
   depends_on "autoconf" => :build
+  depends_on "autoconf-archive" => :build if build.head?
```
Comment (MikeMcQuaid) [inline Formula/mu.rb]: Create a `head do` block and put the `url "https://github.com/djcb/mu.git"` and `depends_on "autoconf-archive" => :build` in there.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/mu.rb (hunk 1 above)
Comment (junkw) [inline Formula/mu.rb]: Thanks for your review.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Comment by junkw** [comment]
Thanks for your review again and the merge

**Pull request merged**